packages/msgraph/src/Adaptor.js

```diff
@@ -76,14 +85,195 @@ export function create(resource, data, callback) {
+ */
+export function getFolder(pathOrId, options, callback = s => s) {
+  return async state => {
+    const defaultOptions = {
+      driveName: 'default', // Named drive in state.drives
+      metadata: false, // If false return folder files if true return folder metadata
+      $filter: '', // Eg: "file/mimeType eq \'application/vnd.ms-excel\'"
+    };
+    const { accessToken, apiVersion } = state.configuration;
+    const [resolvedPathOrId, resolvedOptions] = expandReferences(
+      state,
+      pathOrId,
+      options
+    );
+
+    const { driveName, metadata } = { ...defaultOptions, ...resolvedOptions };
+
+    if (!state.drives[driveName]) {
+      const egJobCode = [
+        'Eg: Get a site drive then get "/Sample Data" folder',
+        'getDrive({ id: "openfn.sharepoint.com", owner: "sites"})',
+        'getFolder("/Sample Data")',
+      ].join('\n\t  ');
+
+      const errorString = [
+        `Message: Drive is not defined`,
+        `Quick Fix: Add getDrive() operation before getFolder()`,
+        `${egJobCode}`,
+      ].join('\n\t∟ ');
+
+      throw new Error(errorString);
+    }
+
+    const { id: driveId } = state.drives[driveName];
+
+    let urlPath;
+
+    if (resolvedPathOrId.startsWith('/')) {
+      urlPath = `drives/${driveId}/root:/${encodeURIComponent(
+        resolvedPathOrId
+      )}`;
+      metadata ? urlPath : (urlPath = `${urlPath}:/children`);
+    } else {
+      urlPath = `drives/${driveId}/items/${resolvedPathOrId}`;
+      metadata ? urlPath : (urlPath = `${urlPath}/children`);
+    }
+
+    const url = getUrl(urlPath, apiVersion);
+    const auth = getAuth(accessToken);
+
+    return request(url, { ...auth }).then(response =>
+      handleResponse(response, state, callback)
+    );
+  };
+}
+
+/**
+ * Get file metadata or file content.
+ * @public
+ * @example <caption>Get a file by ID</caption>
+ * getFile('01LUM6XOGRONYNTZ26DBBJPTN5IFTQPBIW')
+ * @example <caption>Get a file for a named drive by id</caption>
+ * getFile("01LUM6XOGRONYNTZ26DBBJPTN5IFTQPBIW",{ driveName: "mydrive"})
+ * @param {string} pathOrId - A path to a file or file id
+ * @param {object} options - (Optional) Query parametes
+ * @param {function} [callback = s => s] (Optional) Callback function
+ * @return {Operation}
+ */
+export function getFile(pathOrId, options, callback = s => s) {
+  return async state => {
+    const defaultOptions = {
+      driveName: 'default', // named drive in state.drives
+      metadata: false, // Returns file msgraph metadata
+      $filter: '', // Eg: "file/mimeType eq \'application/vnd.ms-excel\'"
+      select: '', // Eg: id,@microsoft.graph.downloadUrl
+    };
+    const { accessToken, apiVersion } = state.configuration;
+    const [resolvedPathOrId, resolvedOptions] = expandReferences(
+      state,
+      pathOrId,
+      options
+    );
+
+    const { driveName, metadata, select } = {
+      ...defaultOptions,
+      ...resolvedOptions,
+    };
+
+    if (!state.drives[driveName]) {
```

Comment:
This whole code block is re-used - it should be factored out into a function (probably in Utils.js?)